tests/test_cli/test_provision.py

```diff
@@ -28,6 +30,15 @@ def mock_provisioned_remote_state():
         yield
 
 
+@pytest.fixture(autouse=True)
+def mock_use_lock():
```

Comment:
Question: Trying to understand why is there a different approach for testing `mock_use_lock.assert_called_once()` in test_provision vs `mock_provisioned_remote_state.use_lock.assert_called_once()` in `test_destory` and `test_get`.

Is it because `mock_state_manager.is_state_provisioned.return_value` is set to False in provision and True in other two test?

Reply:
Yeah, I just followed what was already there in those test files. In get and destroy the whole manager was mocked, and in provision only `is_state_provisioned` was mocked (as provision test cases actually rely on remote state being initialised)